src/util/auth.cpp

```diff
@@ -135,3 +135,11 @@ bool decode_srp_verifier_and_salt(const std::string &encoded,
 	return true;
 
 }
+
+/// Override every character before clearing
+void clear_string(std::string &text)
+{
+	for (size_t i = 0; i < text.size(); i++)
+		text[i] = 0;
```

Comment:
I don't think this necessarily works either. Compilers are smart and will optimize out things you don't expect.
why not use `explicit_bzero`?

Reply:
Is `explicit_bzero` available on Windows as well?
It looks like a function from Glibc. So, Visual Studio uses some Microsoft version of libc, right?

Reply:
Looks like the Android build refuses to know explicit_bzero and memset_s as well.

Maybe this has to be applied for Android? https://github.com/Yubico/libfido2/issues/247#issuecomment-1100346118

And some similar for Mac os x: https://www.appsloveworld.com/c/100/30/mac-os-x-equivalent-of-securezeromemory-rtlsecurezeromemory?expand_article=1

Or apply a mac os x solution for every situation if it is safe solution.

Reply:
So solution based on https://pvs-studio.com/en/blog/posts/cpp/0388/ is now in the branch.
It uses `memset_s` or `SecureZeroMemory` if available and volatile if no one is available.